<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1027</width>
    <height>889</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <property name="sizeConstraint">
     <enum>QLayout::SetDefaultConstraint</enum>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QWidget" name="widEditor" native="true">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTabWidget" name="tabResult">
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <property name="elideMode">
         <enum>Qt::ElideNone</enum>
        </property>
        <widget class="QWidget" name="tabLexer">
         <attribute name="title">
          <string>词法</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QTabWidget" name="tabWidLexer">
            <property name="font">
             <font>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tabWord">
             <attribute name="title">
              <string>符号表</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QTableWidget" name="wordTable">
                <property name="font">
                 <font>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <column>
                 <property name="text">
                  <string>助记符</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>属性值</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>单词符号</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabName">
             <attribute name="title">
              <string>ID 表</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QTableWidget" name="nameTable">
                <property name="font">
                 <font>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <column>
                 <property name="text">
                  <string>ID</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>对象名</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabParser">
         <attribute name="title">
          <string>语法</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QTabWidget" name="tabWidParser">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tabFirst">
             <attribute name="title">
              <string>First表</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="QTableWidget" name="firstTable">
                <property name="font">
                 <font>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <column>
                 <property name="text">
                  <string>非终结符</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>First 集</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabLR1">
             <attribute name="title">
              <string>LR(1)分析表</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_9">
              <item>
               <widget class="QTableWidget" name="LRTable">
                <property name="font">
                 <font>
                  <pointsize>11</pointsize>
                 </font>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabStack">
             <attribute name="title">
              <string>分析过程</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <widget class="QTableWidget" name="stackTable">
                <property name="font">
                 <font>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <column>
                 <property name="text">
                  <string>栈</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>输入串</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>状态栈</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabSemantic">
         <attribute name="title">
          <string>语义</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QTabWidget" name="tabWidSemantic">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tabMid">
             <attribute name="title">
              <string>中间代码</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <widget class="QTableWidget" name="midTable">
                <property name="font">
                 <font>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <column>
                 <property name="text">
                  <string>op</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>arg1</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>arg2</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>result</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabOpt">
             <attribute name="title">
              <string>优化代码</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_11">
              <item>
               <widget class="QTableWidget" name="optTable">
                <property name="font">
                 <font>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <column>
                 <property name="text">
                  <string>op</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>arg1</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>arg2</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>result</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabGenerator">
         <attribute name="title">
          <string>目标</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_14">
          <item>
           <widget class="QTabWidget" name="tabWidGenerator">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tabAct">
             <attribute name="title">
              <string>待用活跃表</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_15">
              <item>
               <widget class="QTableWidget" name="actTable">
                <column>
                 <property name="text">
                  <string>四元式</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>左值</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>左操作数</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>右操作数</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabReg">
             <attribute name="title">
              <string>寄存器分配</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_13">
              <item>
               <widget class="QTableWidget" name="regTable">
                <column>
                 <property name="text">
                  <string>四元式</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>目标代码</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>RVALUE</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>AVALUE</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabObj">
             <attribute name="title">
              <string>MIPS代码</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_12">
              <item>
               <widget class="QTextBrowser" name="objText"/>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label">
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>信息栏</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTextBrowser" name="Message">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>100</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="iconSize">
    <size>
     <width>50</width>
     <height>50</height>
    </size>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actOpen"/>
   <addaction name="actNew"/>
   <addaction name="actSave"/>
   <addaction name="actSaveas"/>
   <addaction name="separator"/>
   <addaction name="actCopy"/>
   <addaction name="actCut"/>
   <addaction name="actPaste"/>
   <addaction name="separator"/>
   <addaction name="actUndo"/>
   <addaction name="actRedo"/>
   <addaction name="separator"/>
   <addaction name="actBegin"/>
   <addaction name="separator"/>
   <addaction name="actQuit"/>
  </widget>
  <action name="actOpen">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/Open.png</normaloff>:/res/Open.png</iconset>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
   <property name="toolTip">
    <string>打开文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actNew">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/New.png</normaloff>:/res/New.png</iconset>
   </property>
   <property name="text">
    <string>新建</string>
   </property>
   <property name="toolTip">
    <string>新建文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actSave">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/Save.png</normaloff>:/res/Save.png</iconset>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="toolTip">
    <string>保存文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actSaveas">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/SaveAs.png</normaloff>:/res/SaveAs.png</iconset>
   </property>
   <property name="text">
    <string>另存为</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actQuit">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/Quit.png</normaloff>:/res/Quit.png</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="toolTip">
    <string>退出程序</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actUndo">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/Undo.png</normaloff>:/res/Undo.png</iconset>
   </property>
   <property name="text">
    <string>撤销</string>
   </property>
   <property name="toolTip">
    <string>撤销上一步</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actRedo">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/Redo.png</normaloff>:/res/Redo.png</iconset>
   </property>
   <property name="text">
    <string>重写</string>
   </property>
   <property name="toolTip">
    <string>重写上一步</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actCopy">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/Copy.png</normaloff>:/res/Copy.png</iconset>
   </property>
   <property name="text">
    <string>复制</string>
   </property>
   <property name="toolTip">
    <string>复制文本</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actCut">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/Cut.png</normaloff>:/res/Cut.png</iconset>
   </property>
   <property name="text">
    <string>剪切</string>
   </property>
   <property name="toolTip">
    <string>剪切文本</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actPaste">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/Paste.png</normaloff>:/res/Paste.png</iconset>
   </property>
   <property name="text">
    <string>粘贴</string>
   </property>
   <property name="toolTip">
    <string>粘贴文本</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actBegin">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/Begin.png</normaloff>:/res/Begin.png</iconset>
   </property>
   <property name="text">
    <string>分析</string>
   </property>
   <property name="toolTip">
    <string>开始分析</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections/>
</ui>
